### ISSUE TITLE
Naming consistency with respect to the Weka lib

### ISSUE THREAD

**Issue by paulgoetze** [arrigonialberto86/ruby-band#8]
Are there deeper thoughts behind using singular names for the Weka modules as `Weka::Filter`, `Weka::Clusterer` and `Weka::Classifier` instead of the plural names used in the original weka library `Weka::Filters`, `Weka::Clusterers` and `Weka::Classifiers`? (Whereas they are placed in plural named directories.)

For me, plural forms would make absolutely sense since there are e.g. several classifier modules and classes in the `Weka::Classifier` module. Consistency might also help in refering to the weka documentation. 

Maybe then also the `<Part>_utils` modules should be nested in a `Weka::<Part>` module, so their name can be a simple `Utils`.

What do you think?

**Comment by arrigonialberto86** [comment]
I completely agree with these observations, and I welcome your suggestions. I started out GSOC last year by trying to introduce a "new" naming system which could (at least in theory) homogeneously integrate different libraries, so the main point was to create a central 'core' supporting datatypes from different sources. Nevertheless, I found out that users (often coming from Java) prefer to stick to the more conventional Weka naming system :) 
So far, I have only added the basics of Apache Math, but I am considering further developments.
In the meantime, any suggestion or change to the API structure (or DSL) is more than welcome :)
